sechub-administration/src/main/java/com/mercedesbenz/sechub/domain/administration/project/ProjectService.java

```diff
@@ -0,0 +1,62 @@
+package com.mercedesbenz.sechub.domain.administration.project;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Service;
+
+import com.mercedesbenz.sechub.domain.administration.user.User;
+import com.mercedesbenz.sechub.domain.administration.user.UserRepository;
+import com.mercedesbenz.sechub.sharedkernel.validation.UserInputAssertion;
+
+@Service
+public class ProjectService {
+
+    private final UserRepository userRepository;
+    private final UserInputAssertion userInputAssertion;
+
+    public ProjectService(UserRepository userRepository, UserInputAssertion userInputAssertion) {
+        this.userRepository = userRepository;
+        this.userInputAssertion = userInputAssertion;
+    }
+
+    public ProjectData[] getProjectDataList(String userId) {
+        userInputAssertion.assertIsValidUserId(userId);
+
+        User user = userRepository.findOrFailUser(userId);
+        List<ProjectData> projects = new ArrayList<>();
+
+        collectProjectDataForUser(user, projects);
```

Comment:
Since the `projects` collection is empty I'd rather have a statement like this

`List<ProjectData> projects = collectProjectDataForUser(user)`

The method creates and returns the collection, and the caller will store it in the projects variable.

This is generally a more declarative style, making the code easier to understand. It clearly indicates that `collectProjectDataForUser` is responsible for returning a new collection based on the user, without modifying any external state.

This is a better choice if you don't want the method to depend on external mutable state (like modifying an existing list) and if you prefer a method to return fresh data instead of updating passed parameters.